src/repositories/tasks.ts

```diff
@@ -0,0 +1,41 @@
+import { OAuth1Client } from '@extrahorizon/javascript-sdk';
+import { permissionModes } from '../commands/tasks/taskConfig';
+
+export interface FunctionCreation {
+  name: string;
+  description: string;
+  entryPoint: string;
+  code: string;
+  runtime: string;
+  timeLimit?: number;
+  memoryLimit?: number;
+  environmentVariables?: Record<string, any>;
+  executionOptions?: {
+    permissionMode: permissionModes;
+  };
+  retryPolicy?: {
+    enabled: boolean;
+    errorsToRetry: string[];
+  };
+}
+
+export const functions = ({
+  async find(sdk: OAuth1Client) {
```

Comment:
(silly, no need to change) Would it make sense call this `list` or `listAll`? Feels a **little** off that `find` and `findByName` don't both return an array

Reply:
This again matches the existing interfaces of the SDK for how the schemas work, I'm open to changing it if it's the consensus... 🤔